pkg/resourcecollector/resourcecollector.go

```diff
@@ -444,11 +446,57 @@ func (r *ResourceCollector) prepareResourcesForCollection(
 	return nil
 }
 
+func isCSIPersistentVolumeClaim(pvcObject runtime.Unstructured, allObjects []runtime.Unstructured) (bool, error) {
+	pvc := v1.PersistentVolumeClaim{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(pvcObject.UnstructuredContent(), &pvc); err != nil {
+		return false, fmt.Errorf("error converting to persistent volume claim: %v", err)
+	}
+	pvName := pvc.Spec.VolumeName
+
+	// Must check PV object to see if it is a CSI volume or not
+	// PV object corresponding to this PVC should also be in resources.json
+	for _, o := range allObjects {
+		objectType, err := meta.TypeAccessor(o)
+		if err != nil {
+			return false, err
+		}
+
+		// If a PV, check if it's the PV name associated with this PVC
+		if objectType.GetKind() == "PersistentVolume" {
+			pv := v1.PersistentVolume{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(o.UnstructuredContent(), &pv); err != nil {
+				return false, fmt.Errorf("error converting to persistent volume claim: %v", err)
+			}
+
+			if pv.Name == pvName {
+				// Same PV name for PVC. Check if PV is CSI
+				return isCSIPersistentVolume(&pv)
+			}
+		}
+	}
+
+	return false, nil
+}
+
+func isCSIPersistentVolume(pv *v1.PersistentVolume) (bool, error) {
+	if pv.Spec.CSI != nil {
```

Comment:
`GetPVDriver` won't work for this case. In this scenario, we're checking if it's okay to restore the PV and PVC. 

We need to backup the PV/PVC yaml to apply labels/etc upon restore. So, this is intended to skip directly restoring the PV/PVC objects. 

During restore, the PV does not exist yet, so csi.OwnsPVC will not work. This is because I need to get the PV corresponding to a given PVC inside of csi.OwnsPVC.

Reply:
Reworking this to call GetPVDriver for both PV and PVCs. For PVCs, I'll need to find the PV associated with it. 

Had to refactor this around today, as `includeObject` is now called both on backup and restore side.